app/src/main/java/org/wikipedia/page/references/ReferenceDialog.java

```diff
@@ -26,42 +30,58 @@
 
 import butterknife.BindView;
 import butterknife.ButterKnife;
+import butterknife.Unbinder;
 
 import static org.wikipedia.util.L10nUtil.setConditionalLayoutDirection;
 
 /**
  * A dialog that displays the currently clicked reference.
  */
-public class ReferenceDialog extends BottomSheetDialog {
+public class ReferenceDialog extends ExtendedBottomSheetDialogFragment {
+    public interface Callback {
+        LinkHandler getLinkHandler();
+        List<PageReferences.Reference> getReferencesGroup();
+        int getSelectedReferenceIndex();
+    }
+
     @BindView(R.id.reference_pager) ViewPager2 referencesViewPager;
     @BindView(R.id.page_indicator_view) TabLayout pageIndicatorView;
     @BindView(R.id.indicator_divider) View pageIndicatorDivider;
     @BindView(R.id.reference_title_text) TextView titleTextView;
-    private LinkHandler referenceLinkHandler;
+    private Unbinder unbinder;
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_references_pager, null);
+        unbinder = ButterKnife.bind(this, rootView);
+
+        titleTextView.setText(requireContext().getString(R.string.reference_title, ""));
+        referencesViewPager.setOffscreenPageLimit(2);
+        referencesViewPager.setAdapter(new ReferencesAdapter(callback().getReferencesGroup()));
+        new TabLayoutMediator(pageIndicatorView, referencesViewPager, (tab, position) -> { }).attach();
+        referencesViewPager.setCurrentItem(callback().getSelectedReferenceIndex(), true);
 
-    public ReferenceDialog(@NonNull Context context, int selectedIndex, List<PageReferences.Reference> adjacentReferences, LinkHandler referenceLinkHandler) {
-        super(context);
-        View rootView = LayoutInflater.from(context).inflate(R.layout.fragment_references_pager, null);
-        setContentView(rootView);
-        ButterKnife.bind(this);
-        this.referenceLinkHandler = referenceLinkHandler;
+        setConditionalLayoutDirection(rootView, callback().getLinkHandler().getWikiSite().languageCode());
+        return rootView;
+    }
 
-        if (adjacentReferences.size() == 1) {
+    @Override
+    public void onStart() {
+        super.onStart();
+        if (callback().getReferencesGroup().size() == 1) {
             pageIndicatorView.setVisibility(View.GONE);
             ((ViewGroup) pageIndicatorView.getParent()).removeView(pageIndicatorView);
```

Comment:
If we try to switch to the other app and back, the app will crash.

`void android.view.ViewGroup.removeView(android.view.View)' on a null object reference`

Reply:
Good catch, I don't know why we have that line anyway; it's not doing anything useful.